obspy/core/stream.py

```diff
@@ -1797,6 +1822,38 @@ def select(self, network=None, station=None, location=None, channel=None,
         All other selection criteria that accept strings (network, station,
         location) may also contain Unix style wildcards (``*``, ``?``, ...).
         """
+        if inventory is not None:
+            trace_ids = []
+            start_dates = []
+            end_dates = []
+            for net in inventory.networks:
+                for sta in net.stations:
+                    for chan in sta.channels:
+                        id = '.'.join((net.code, sta.code,
+                                       chan.location_code, chan.code))
+                        trace_ids.append(id)
+                        start_dates.append(chan.start_date)
+                        end_dates.append(chan.end_date)
+            traces = []
+            for trace in self:
+                idx = 0
+                while True:
+                    try:
+                        idx = trace_ids.index(trace.id, idx)
+                        start_date = start_dates[idx]
+                        end_date = end_dates[idx]
+                        idx += 1
+                        if start_date is not None and\
+                                trace.stats.starttime < start_date:
+                            continue
+                        if end_date is not None and\
+                                trace.stats.endtime > end_date:
+                            continue
+                        traces.append(trace)
+                    except ValueError:
+                        break
+            return self.__class__(traces=traces)
```

Comment:
If I understand correctly any other options specified to `select()` will be ignored if an Inventory is provided?

Reply:
Correct. It seems to me a strange use case to mix the two criteria. What do you think?

Reply:
I think we should stick with the current approach of chaining all checks with logical AND, i.e. any filter criteria specified by the user should actually also be applied. I don't think it's good practice to have one kwarg completely override others too much, it'd be hell for a user to check which combination of kwargs together actually do what.

Reply:
I shared @megies opinion. This would be really confusing to me and one passes multiple criteria I would intuitively expect them all to be applied.

Reply:
Ok, I'll modify that.

Reply:
Please try to get it done ASAP so I can bundle the RC thanks! :-)

Reply:
Did it myself. Tests aren't comprehensive, btw, no fails popping up after my change it seems